qbg/ecp22.c

```diff
@@ -943,6 +943,7 @@ static int ecp22_req2send(char *ifname, unsigned short subtype,
 		goto out;
 	}
 	if (ptlv->size >= ECP22_MAXPAYLOAD_LEN) {
+		free_pkd_tlv(ptlv);
```

Comment:
This check and the `!ptlv` guard just above this should probably be moved to the top of the file as they are effectively checking the return value of `copy_ptlv(du)` which is the first code actually called in this function. This is also needed to fix the same memory leak in the `!ecp` guard that occurs above which should also call `free_pkd_tlv(ptlv)` on error, but only if `ptlv != NULL` which is only checked for afterwards in the current logic.
```suggestion
	if (!ptlv) {
		rc = -ENOMEM;
		goto out;
	}
	if (ptlv->size >= ECP22_MAXPAYLOAD_LEN) {
		free_pkd_tlv(ptlv);
		rc = -E2BIG;
		goto out;
	}

	eud = find_module_user_data_by_id(&lldp_mod_head, LLDP_MOD_ECP22);
	ecp = find_ecpdata(ifname, eud);
	if (!ecp) {
		free_pkd_tlv(ptlv);
		rc = -ENODEV;
		goto out;
	}
```